src/optimizer/plugins/ResolveValueRuleURLsOptimizer.ts

```diff
@@ -28,13 +28,15 @@ export default {
           (rule instanceof ExportableAssignmentRule || rule instanceof ExportableCaretValueRule) &&
           rule.value instanceof fshtypes.FshCode
         ) {
-          rule.value.system =
-            resolveAliasFromURL(rule.value.system, pkg.aliases) ?? rule.value.system;
+          rule.value.system = optimizeURL(
+            rule.value.system,
+            pkg.aliases,
+            [utils.Type.CodeSystem],
+            fisher,
+            options.alias ?? true
+          );
```

Comment:
I love that we're re-using this, but it may be too aggressive. The request was to use _local_ code system names instead of aliases -- but this replaces _all_ code system URLs with their names (including those from core FHIR and dependency IGs). As an example, here is a screenshot of a diff of the aliases in US Core between `master` and this branch:

<img width="2533" alt="image" src="https://user-images.githubusercontent.com/2278253/231896097-9c4bb174-53bd-4758-a900-3849739cf788.png">

We probably want to think about if this is what we really want. I think best practice is generally to use URLs for external systems and names for local systems. That's the direction I'm leaning, but if you disagree then I think it's worth a discussion!

Reply:
I think you're right: this change might be too aggressive in reducing the number of aliases. I'll try to adjust this to use names only for systems in the lake.